src/client/dfs/dfs.c

```diff
@@ -636,13 +638,16 @@ entry_stat(dfs_t *dfs, daos_handle_t th, daos_handle_t oh, const char *name,
 
 	/* Check if parent has the entry */
 	rc = fetch_entry(oh, th, name, len, true, &exists, &entry,
-			 0, NULL, NULL, NULL);
+			 xnr, xnames, xvals, xsizes);
 	if (rc)
 		return rc;
 
 	if (!exists)
 		return ENOENT;
 
+	if (eoid && (eoid->hi != entry.oid.hi || eoid->lo != entry.oid.lo))
```

Comment:
obj is only used in this function for checkout the expected oid so this function doesn't need to take and check both, it's probably better to keep eiod rather than obj however.

Reply:
unless you are really objecting about this, i'd rather land this now and do this in a future PR.

Reply:
it's going to be a reasonably large change to dfuse to use this so there's no a particular hurry, I'm not really objecting to this however.